### DISCUSSION TITLE
VSCode JSON formatter

### DISCUSSION THREAD

**Discussion by MohanVallivedu** [microsoft/vscode-discussions#434] [Extension Development QnA] [answered]
Hello all,
I am wondering if there is any setting for JSON formatter in VSCode (Not json.format.keepLines) to be able to keep an array in a single line.
JSON expected format:
```
{
  "schemas": {
    "required": ["name"],
    "oneOf": [{
         "$ref": "#/components/schemas/NotFound"
      }, {
         "$ref": "#/components/schemas/ApiNotFound"
      }]
   }
}
```
The present VSCode JSON formatter will format as below:
```
{
  "schemas": {
    "required": [
        "name"
     ],
     "oneOf": [
           {
              "$ref": "#/components/schemas/NotFound"
           },
           {
               "$ref": "#/components/schemas/ApiNotFound"
           }
     ]
  }
}
```

**Comment by metatron1973** [comment]
Nice

**Comment by isidorn** [comment]
Maybe @aiday-mar can help

**Comment by aiday-mar** [comment, marked as answer]
Hi, thanks for your comment. I do not think there is such a functionality as of yet that formats the whole document without formatting the arrays. As you mentioned there is the keepLines feature as well as the standard formatting.
